sequencer/finalizer.go

```diff
@@ -392,36 +392,36 @@ func (f *finalizer) processTransaction(ctx context.Context, tx *TxTracker) error
 	f.batch.localExitRoot = processBatchResponse.NewLocalExitRoot
 	log.Infof("processTransaction: data loaded in memory. batch.batchNumber: %d, batchNumber: %d, result.NewStateRoot: %s, result.NewLocalExitRoot: %s, oldStateRoot: %s", f.batch.batchNumber, f.processRequest.BatchNumber, processBatchResponse.NewStateRoot.String(), processBatchResponse.NewLocalExitRoot.String(), oldStateRoot.String())
 
-	return nil
+	return nil, nil
 }
 
 // handleProcessTransactionResponse handles the response of transaction processing.
-func (f *finalizer) handleProcessTransactionResponse(ctx context.Context, tx *TxTracker, result *state.ProcessBatchResponse, oldStateRoot common.Hash) error {
+func (f *finalizer) handleProcessTransactionResponse(ctx context.Context, tx *TxTracker, result *state.ProcessBatchResponse, oldStateRoot common.Hash) (errWg *sync.WaitGroup, err error) {
 	// Handle Transaction Error
+
 	errorCode := executor.RomErrorCode(result.Responses[0].RomError)
-	if !state.IsStateRootChanged(errorCode) {
+	if result.IsRomOOCError || executor.IsIntrinsicError(errorCode) {
```

Comment:
I think you try to make this more performant, but I think is better and mor clear using the function.

Reply:
okay will revert it